### PR TITLE
Fix for s390x failure

### LINKED ISSUE #13206: 5.1RC1: test_fit_with_fixed_and_bound_constraints[DogBoxLSQFitter] fails on s390x
### Description

After applying After applying ec4a8f7fb1a3420fd4aceff21dc38d2c331fa6e3, I get the following error:

```__________ test_fit_with_fixed_and_bound_constraints[DogBoxLSQFitter] __________

fitter = <astropy.modeling.fitting.DogBoxLSQFitter object at 0x3ff97821cc0>

    @pytest.mark.skipif('not HAS_SCIPY')
    @pytest.mark.filterwarnings(r'ignore:divide by zero encountered.*')
    @pytest.mark.parametrize('fitter', fitters)
    def test_fit_with_fixed_and_bound_constraints(fitter):
        """…"""
        fitter = fitter()
    
        m = models.Gaussian1D(amplitude=3, mean=4, stddev=1,
                              bounds={'mean': (4, 5)},
                              fixed={'amplitude': True})
        x = np.linspace(0, 10, 10)
        y = np.exp(-x ** 2 / 2)
    
>       fitted_1 = fitter(m, x, y)

astropy/modeling/tests/test_constraints.py:401: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
astropy/modeling/fitting.py:269: in wrapper
    return func(self, model, x, y, z=z, **kwargs)
astropy/modeling/fitting.py:1222: in __call__
    init_values, fitparams, cov_x = self._run_fitter(model_copy, farg,
astropy/modeling/fitting.py:1364: in _run_fitter
    self.fit_info = optimize.least_squares(
/usr/lib/python3/dist-packages/scipy/optimize/_lsq/least_squares.py:939: in least_squares
    result = dogbox(fun_wrapped, jac_wrapped, x0, f0, J0, lb, ub, ftol,
/usr/lib/python3/dist-packages/scipy/optimize/_lsq/dogbox.py:222: in dogbox
    newton_step = lstsq(J_free, -f, rcond=-1)[0]
<__array_function__ internals>:5: in lstsq
    ???
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

a = array([[0.00000000e+000],
       [1.73963948e-175],
       [1.33461258e-064],
       [3.77464664e-007],
       [3.3497...3419468e-049],
       [4.70443110e-149],
       [1.84469454e-302],
       [0.00000000e+000],
       [0.00000000e+000]])
b = array([[ 1.00000000e+00],
       [ 5.39407507e-01],
       [ 8.46579886e-02],
       [ 3.86591928e-03],
       [-1.211...98575042e-07],
       [ 2.23363144e-10],
       [ 7.31025182e-14],
       [ 6.96124846e-18],
       [ 1.92874985e-22]])
rcond = -1

    @array_function_dispatch(_lstsq_dispatcher)
    def lstsq(a, b, rcond="warn"):
        r"""…"""
        a, _ = _makearray(a)
        b, wrap = _makearray(b)
        is_1d = b.ndim == 1
        if is_1d:
            b = b[:, newaxis]
        _assert_2d(a, b)
        m, n = a.shape[-2:]
        m2, n_rhs = b.shape[-2:]
        if m != m2:
            raise LinAlgError('Incompatible dimensions')
    
        t, result_t = _commonType(a, b)
        # FIXME: real_t is unused
        real_t = _linalgRealType(t)
        result_real_t = _realType(result_t)
    
        # Determine default rcond value
        if rcond == "warn":
            # 2017-08-19, 1.14.0
            warnings.warn("`rcond` parameter will change to the default of "
                          "machine precision times ``max(M, N)`` where M and N "
                          "are the input matrix dimensions.\n"
                          "To use the future default and silence this warning "
                          "we advise to pass `rcond=None`, to keep using the old, "
                          "explicitly pass `rcond=-1`.",
                          FutureWarning, stacklevel=3)
            rcond = -1
        if rcond is None:
            rcond = finfo(t).eps * max(n, m)
    
        if m <= n:
            gufunc = _umath_linalg.lstsq_m
        else:
            gufunc = _umath_linalg.lstsq_n
    
        signature = 'DDd->Ddid' if isComplexType(t) else 'ddd->ddid'
        extobj = get_linalg_error_extobj(_raise_linalgerror_lstsq)
        if n_rhs == 0:
            # lapack can't handle n_rhs = 0 - so allocate the array one larger in that axis
            b = zeros(b.shape[:-2] + (m, n_rhs + 1), dtype=b.dtype)
>       x, resids, rank, s = gufunc(a, b, rcond, signature=signature, extobj=extobj)
E       ValueError: On entry to DLASCL parameter number 4 had an illegal value

/usr/lib/python3/dist-packages/numpy/linalg/linalg.py:2306: ValueError
```

I *think* this is unrelated, therefore the new issue.

### System Details
latform: Linux-5.10.0-13-s390x-s390x-with-glibc2.33

Executable: /usr/bin/python3.10

Full Python Version: 
3.10.4 (main, Apr  2 2022, 09:04:19) [GCC 11.2.0]

encodings: sys: utf-8, locale: UTF-8, filesystem: utf-8
byteorder: big
float info: dig: 15, mant_dig: 15

Package versions: 
Numpy: 1.21.5
Scipy: 1.7.3
Matplotlib: 3.5.1
h5py: 3.6.0
PyERFA: 2.0.0.1
Cython: 0.29.28
Scikit-image: 0.18.3

### PR DESCRIPTION
**Pull request by WilliamJamieson**
### Description

Fixes #13206

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [x] Do the proposed changes actually accomplish desired goals?
- [x] Do the proposed changes follow the [Astropy coding guidelines](https://docs.astropy.org/en/latest/development/codeguide.html)?
- [x] Are tests added/updated as required? If so, do they follow the [Astropy testing guidelines](https://docs.astropy.org/en/latest/development/testguide.html)?
- [x] Are docs added/updated as required? If so, do they follow the [Astropy documentation guidelines](https://docs.astropy.org/en/latest/development/docguide.html#astropy-documentation-rules-and-guidelines)?
- [x] Is rebase and/or squash necessary? If so, please provide the author with appropriate instructions. Also see ["When to rebase and squash commits"](https://docs.astropy.org/en/latest/development/when_to_rebase.html).
- [x] Did the CI pass? If no, are the failures related? If you need to run daily and weekly cron jobs as part of the PR, please apply the `Extra CI` label.
- [x] Is a change log needed? If yes, did the change log check pass? If no, add the `no-changelog-entry-needed` label. If this is a manual backport, use the `skip-changelog-checks` label unless special changelog handling is necessary.
- [x] Is this a big PR that makes a "What's new?" entry worthwhile and if so, is (1) a "what's new" entry included in this PR and (2) the "whatsnew-needed" label applied?
- [x] Is a milestone set? Milestone must be set but `astropy-bot` check might be missing; do not let the green checkmark fool you.
- [x] At the time of adding the milestone, if the milestone set requires a backport to release branch(es), apply the appropriate `backport-X.Y.x` label(s) *before* merge.

### CODE REVIEW

**Comment by WilliamJamieson** [comment]
@olebole is there a way to tell if the architecture is s390x in python?

**Comment by olebole** [comment]
There is for sure something in the `platform` module, f.e. `platform.machine()`; however I would hesitate a bit to be too platform specific here -- in the meantime, I also saw [ARM 32 bin soft float little endian failing](https://buildd.debian.org/status/fetch.php?pkg=astropy&arch=armel&ver=5.1%7Erc1-2&stamp=1651603413&raw=0).

File: astropy/modeling/tests/test_constraints.py
Comment (olebole) [inline astropy/modeling/tests/test_constraints.py:395]: Maybe it is better to use `pytest.xfail()`?

**Comment by WilliamJamieson** [comment]
> There is for sure something in the `platform` module, f.e. `platform.machine()`; however I would hesitate a bit to be too platform specific here -- in the meantime, I also saw [ARM 32 bin soft float little endian failing](https://buildd.debian.org/status/fetch.php?pkg=astropy&arch=armel&ver=5.1%7Erc1-2&stamp=1651603413&raw=0).

Is this failing with respect to this PR or the 5.1 release candidate?

**Comment by olebole** [comment]
On armel, it fails with #13206 on 5.1RC, both with and without #13204 applied.

**Comment by WilliamJamieson** [comment]
@olebole d341fcc has platform specific xfails for this test. Can you check that I got them right and confirm this fixes your issues?

**Comment by olebole** [comment]
I am not really happy with platform specific xfails, since they seem to appear occasionally; so we may not have catched all flaky platforms yet.
I would prefer to generally have an xfail here as long as we don't know the exact (platform specific) clause.

**Comment by WilliamJamieson** [comment]
> I am not really happy with platform specific xfails, since they seem to appear occasionally; so we may not have catched all flaky platforms yet. I would prefer to generally have an xfail here as long as we don't know the exact (platform specific) cause.

I disagree with a general xfail for a number of reasons. This is a different type of failure than what we were seeing in #13202. Here the failure is originating within the scipy algorithm itself; moreover, the fitting problem itself should be well behaved as we are fitting a `Gaussian1D` model to Gaussian data (so it should be numerically stable). Finally, this is one of the only tests for `DogBoxLSQFitter` which explicitly tests bound constrained fitting using that fitter. Since it passes on "normal" platforms I feel it is better to have it actually run the tests to get an indication that this feature is "working" rather than have it xfail due to issues on another platform (which could make any future bugs silent with respect to the tests).

**Comment by WilliamJamieson** [comment]
@olebole I changed the platform filter so that it would only work on `arm64` and `x86_64` as you requested. Do these changes allow you to build on all your test platforms now?

**Review by olebole — CHANGES_REQUESTED** [review]
I got now successful builds for [s390x](https://buildd.debian.org/status/fetch.php?pkg=astropy&arch=s390x&ver=5.1%7Erc1-3&stamp=1651686240&raw=0), [armel](https://buildd.debian.org/status/fetch.php?pkg=astropy&arch=armel&ver=5.1%7Erc1-3&stamp=1651690797&raw=0), [mips64el](https://buildd.debian.org/status/fetch.php?pkg=astropy&arch=mips64el&ver=5.1%7Erc1-3&stamp=1651696410&raw=0) (and all other supported architectures).

Before merging, I would suggest to squash all commits into a single one to make the commit log more readable on the long term.

File: astropy/modeling/tests/test_constraints.py
Comment (pllim) [inline astropy/modeling/tests/test_constraints.py:396]: Can you give this xfail a reason? It could be as simple as referring to the original issue so we have a trail. Thanks!

Comment (WilliamJamieson) [reply astropy/modeling/tests/test_constraints.py]: See 495cbf6.

**Review by pllim — APPROVED** [review]
Thanks! The diff LGTM. I'll merge if CI passes.

**Comment by pllim** [comment]
@meeseeksdev backport to v5.1.x

**Comment by olebole** [comment]
@WilliamJamieson Thank you very much for your quick reaction, the communication, and the fix! :+1:

**Pull request merged**
